### PR TITLE
Add component to display example code

### PR DESCRIPTION
**Pull request by mrgnr**
## Done

- Add component to display example code

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/4320

## QA

- Open [demo](https://vanilla-framework-4321.demos.haus/docs/patterns/notification)
- Verify that the code is updated when different options are selected

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [ ] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

File: templates/docs/examples/patterns/notifications/toast.html [hunk 1]
```diff
@@ -11,26 +11,26 @@
 {% set timestamp = request.args.get('timestamp') %}
 
 {% block content %}
-<div class="p-notification{{ '--' ~ type if type else '' }} {% if theme == 'dark' %}is-dark{% endif %} {% if style == 'inline' %}is-inline{% endif %}">
+<div class="p-notification{{ '--' ~ type if type else '' }}{% if theme == 'dark' %} is-dark{% endif %}{% if style == 'inline' %} is-inline{% endif %}">
   <div class="p-notification__content">
     <h5 class="p-notification__title">Permissions changed</h5>
     <p class="p-notification__message">Anyone with access can view your invited users.</p>
-    {% if dismiss == 'true' %}
+    {%- if dismiss == 'true' %}
```
Comment (carkod) [inline templates/docs/examples/patterns/notifications/toast.html:18]: What are these dashes for?

File: templates/docs/examples/patterns/notifications/toast.html [hunk 2]
```diff
@@ -11,26 +11,51 @@
...
+  });
+}
+
+// Set up all notification close buttons.
+var closeButtons = document.querySelectorAll('.p-notification__close');
+
+for (var i = 0, l = closeButtons.length; i < l; i++) {
+  setupCloseButton(closeButtons[i]);
+}
+</script>
```
Comment (minkyngkm) [inline templates/docs/examples/patterns/notifications/toast.html:59]: This script doesn't seem to be working.

Comment (minkyngkm) [reply templates/docs/examples/patterns/notifications/toast.html:59]: @bethcollins92 asked @bartaz about this functionality and it seems to be best to remove this functionality for now! :)

**Review by minkyngkm — APPROVED** [review]
The example code is displaying properly for me! Nice work @mrgnr! 
I just left a comment for the dismiss button. You can remove that script for now :)

File: templates/docs/examples/patterns/notifications/toast.html (hunk 2 above)
Comment (mrgnr) [reply templates/docs/examples/patterns/notifications/toast.html:59]: It's working now, I'd just forgotten to add `id="notification"` on the div :)

**Pull request merged**